ngraph/test/reshape_opt_kernel.cpp

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <numeric>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/runtime/opt_kernel/reshape.hpp"
+#include "ngraph/shape.hpp"
+
+#include "util/ndarray.hpp"
+
+using namespace ngraph;
+
+namespace
+{
+    using ElementValue = int32_t;
+    enum class AxisOrder
+    {
+        straight,
+        reverse,
```

Comment:
Why do we need the reverse order?

Reply:
This allow to parametrize test for two axis order:
straight: {1,2,3,...,n-1}
reverse{n-1,n-2,n-3,..0}